kroxylicious-kms-test-support/src/main/java/io/kroxylicious/kms/service/AbstractTestKmsFacadeTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.service;
+
+import java.util.Objects;
+
+import org.junit.jupiter.api.Test;
+
+import io.kroxylicious.kms.service.TestKekManager.AlreadyExistsException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * Abstract
+ * @param <C> The config type
+ * @param <K> The key reference
+ * @param <E> The type of encrypted DEK
+ */
+@SuppressWarnings("java:S5960") // this is test code, it exists in the main module to facilitate its use by concrete test cases
+public abstract class AbstractTestKmsFacadeTest<C, K, E> {
+
+    private static final String ALIAS = "myalias";
+    protected final TestKmsFacadeFactory<C, K, E> factory;
+
+    protected AbstractTestKmsFacadeTest(TestKmsFacadeFactory<C, K, E> factory) {
+        Objects.requireNonNull(factory);
+        this.factory = factory;
+    }
+
+    @Test
+    void factory() {
+        try (var facade = factory.build()) {
+            assertThat(facade)
+                    .isNotNull()
+                    .extracting(TestKmsFacade::isAvailable)
+                    .isEqualTo(true);
```

Comment:
Can we have this assertion? Doesn't this imply that we have to have `DockerClientFactory.instance().isDockerAvailable()` passing in order to run the build? 

I'm sure that holds for the project development team but feels a bit off as a requirement.

Reply:
If Docker isn't available, io.kroxylicious.kms.provider.hashicorp.vault.VaultTestKmsFacadeTest#beforeEach prevents the test from even starting.   Is that too obscure?

Reply:
Ahh ok, no that works nicely :)